src/bootstrap/build/sanity.rs

```diff
@@ -78,6 +79,24 @@ pub fn check(build: &mut Build) {
             panic!("the iOS target is only supported on OSX");
         }
 
+        if cfg!(windows) {
```

Comment:
Sorry I think I'm not being clear enough. So if you're liking with an external LLVM that is itself dynamically linked, then the only hope we have of working correctly is to ensure that the directory with that LLVM dynamic library is in the runtime linker lookup path. On Windows this just happens to be `PATH` but on Linux/OSX this id `LD_LIBRARY_PATH` and `DYLD_LIBRARY_PATH`.

Which is to say, if we link to a dynamically linked LLVM on *any* platform, we need to run this logic to ensure that the relevant directory is in the relevant environment variable.